### ISSUE TITLE
forward AD for _euclidean_dist

### ISSUE THREAD

**Issue by jparkhill** [pytorch/pytorch#88691]
### 🚀 The feature, motivation and pitch

Submitting this feature request to get functorch.hessians downstream of torch.cdist, and the error issued by pytorch asked me to submit this to prioritize this. 

### Alternatives

I'll try autograd hessian. 

### Additional context

<img width="1035" alt="image" src="https://user-images.githubusercontent.com/4132346/200657073-54ef92cf-9c51-4f4b-9f57-ac47159c5d88.png">

**Comment by zou3519** [comment]
An alternative for `functorch.hessian(f)` is `functorch.jacrev(functorch.jacrev(f))`

**Comment by jparkhill** [comment]
Yes. The autograd hessian works for my purposes too. I am just following orders. Thanks Richard.
